test-it/karma/publicapi/resizeArray.test.js

```diff
@@ -0,0 +1,76 @@
+describe('Arrays in Vireo', function () {
+    'use strict';
+    // Reference aliases
+    var Vireo = window.NationalInstruments.Vireo.Vireo;
+    var vireoRunner = window.testHelpers.vireoRunner;
+    var fixtures = window.testHelpers.fixtures;
+
+    // Sharing Vireo instances across tests make them run soooo much faster
+    var vireo = new Vireo();
+    var viaPath = fixtures.convertToAbsoluteFromTestItDir('ArrayDemo.via');
+    var viName = 'ArrayDemo';
+    var runSlicesAsync;
+
+    beforeEach(function () {
+        runSlicesAsync = vireoRunner.rebootAndLoadVia(vireo, viaPath);
+    });
+
+    it('can be resized when they are variable size 1D array', function (done) {
+        runSlicesAsync(function (rawPrint, rawPrintError) {
+            expect(rawPrint).toBeNonEmptyString();
+            expect(rawPrintError).toBeEmptyString();
+
+            var resized = vireo.eggShell.resizeArray(viName, 'variableArray1d', [5]);
+            console.log(resized);
+            expect(vireo.eggShell.getArrayDimLength(viName, 'variableArray1d', 0)).toBe(5);
+            done();
+        });
+    });
+
+    it('resizes initialized variable 1d array with defaults', function (done) {
+        var variableName = 'variableArray1dwithDefaults';
+        runSlicesAsync(function (rawPrint, rawPrintError) {
+            expect(rawPrint).toBeNonEmptyString();
+            expect(rawPrintError).toBeEmptyString();
+
+            var resized = vireo.eggShell.resizeArray(viName, variableName, [6]);
+
+            expect(resized).toBe(0);
+            expect(vireo.eggShell.getArrayDimLength(viName, variableName, 0)).toBe(6);
+
+            done();
+        });
+    });
+
+    it('cannot resize fixed 2d arrays', function (done) {
+        var variableName = 'fixedArray2d';
+        runSlicesAsync(function (rawPrint, rawPrintError) {
+            expect(rawPrint).toBeNonEmptyString();
+            expect(rawPrintError).toBeEmptyString();
+
+            var resized = vireo.eggShell.resizeArray(viName, variableName, [3, 4]);
+
+            expect(resized).toBe(-2);
```

Comment:
Add another tests when the returned value is -1, that is the array symbol didn't exist.